src/expandableCalendar/agendaListNew.tsx

```diff
@@ -0,0 +1,259 @@
+    }
+  }, 1000, {leading: false, trailing: true}), [ sections]);
+
+  const layoutProvider = useMemo(
+    () => new LayoutProvider(
+      (index) => data[index]?.isTitle ? 'title': 'page',
+      (type, dim) => {
+        dim.width = constants.screenWidth;
+        dim.height = type === 'title' ? 60 : 120;
+      }
+    ),
+    [data]
+  );
+
+  const _onScroll = useCallback((rawEvent: any) => {
+    if (!didScroll.current) {
+      didScroll.current = true;
+      scrollToSection.cancel();
+    }
+
+    // Convert to a format similar to NativeSyntheticEvent<NativeScrollEvent>
+    const event = {
+      nativeEvent: {
+        contentOffset: rawEvent.nativeEvent.contentOffset,
+        layoutMeasurement: rawEvent.nativeEvent.layoutMeasurement,
+        contentSize: rawEvent.nativeEvent.contentSize,
+      },
+    };
+    onScroll?.(event as any);
+  }, [onScroll]);
+
+  const _onVisibleIndicesChanged = useCallback(debounce((all: number[]) => {
+    if (all && all.length && !sectionScroll.current) {
+      const topItemIndex = all[0];
+      const topSection = data[findItemTitleIndex(topItemIndex)];
+      if (topSection && topSection !== _topSection.current) {
+        _topSection.current = topSection.title;
+        if (didScroll.current && !avoidDateUpdates) {
+          // to avoid setDate() on first load (while setting the initial context.date value)
+          setDate?.(topSection.title, UpdateSources.LIST_DRAG);
+        }
+      }
+    }
+  }, 1000, {leading: false, trailing: true},), [avoidDateUpdates, setDate, data]);
+
+  const findItemTitleIndex = useCallback((itemIndex: number) => {
+    let titleIndex = itemIndex;
+    while (titleIndex > 0 && !data[titleIndex]?.isTitle) {
+      titleIndex--;
+    }
+
+    return titleIndex;
+  }, [data]);
+
+  const _onMomentumScrollEnd = useCallback(() => {
+    sectionScroll.current = false;
+  }, []);
+
+  const headerTextStyle = useMemo(() => [style.current.sectionText, sectionStyle], [sectionStyle]);
+
+  const onHeaderLayout = useCallback((event: LayoutChangeEvent) => {
+    sectionHeight.current = event.nativeEvent.layout.height;
+  }, []);
+
+  const _renderSectionHeader = useCallback((info: {section: SectionListData<any, DefaultSectionT>}) => {
+    const title = info?.section?.title;
+
+    if (renderSectionHeader) {
+      return renderSectionHeader(title);
+    }
+
+    const headerTitle = getSectionTitle(title);
+    return <AgendaSectionHeader title={headerTitle} style={headerTextStyle} onLayout={onHeaderLayout}/>;
+  }, [headerTextStyle]);
+
+  const _renderItem = useCallback((_type: any, item: any) => {
+    if (item?.isTitle) {
+      return _renderSectionHeader({section: item});
+    }
+
+    if (props.renderItem) {
+      return props.renderItem({item} as any);
+    }
+
+    return <></>;
+  }, [props.renderItem]);
+
+  const _onEndReached = useCallback(() => {
+    if (props.onEndReached) {
+      props.onEndReached({distanceFromEnd: 0});
```

Comment:
Why distanceFromEnd is hard-coded to 0?

Reply:
The `onEndReached` of `RecyclerListView` is not pass any argument, so I don't have this information

Reply:
It’s required on the onEndReach on the agenda list props, and I don’t want to change existing props

Reply:
It a FlatList prop: onEndReached?: ((info: { distanceFromEnd: number }) => void) | null | undefined; So you can return undefined

Reply:
I want to apply this CB for users that want to know when the end is reached, for applying this CB I must provide an info parameter (it's not optional), the null | undefined part just says that it's not required CB

![image](https://github.com/wix/react-native-calendars/assets/91827012/fb4643b1-65be-4072-b63c-30d2cfe79015)

Reply:
Ok... Add a comment about that please